index.js

```diff
@@ -4,11 +4,14 @@ let hash = crypto.createHash("md5")
 
 let history = {}
 
-function hashify(input, length) {
+function hashify(input, length, tries) {
 	let hashed = hash.update(input).digest("hex").slice(0, length)
 
 	if (history[hashed] && history[hashed] !== input) {
-		throw new Error("babel-plugin-hash-strings encountered a duplicate key name, please increase the 'chars' length")
+		if (tries > 3) {
```

Comment:
max tries could be configurable

Reply:
The problem with this would be that a name could change between versions unexpectedly.

V1

```rb
"string 1" -> "hashA"
"string 2" -> "hashB"
"string 3" -> "hashC"
```

V1

```rb
"string 1" -> "hashA"
"string 2" -> "hashB"
"string 2a" -> "hashC" # added / matched existing key before it came up
"string 3" -> "hashCX" # "hashC" -> "hashCX"
```

Reply:
that makes sense, I thought the hash cache was only per build. In that case I'll close the PR

Reply:
Babel plugins stay in memory for a lot of watch build tools, but I think people wanted to have this plugin emit a mapping of hashes to original strings somehow.